src/Microsoft.SqlTools.Authentication/Sql/AuthenticationProvider.cs

```diff
@@ -27,13 +28,23 @@ public class AuthenticationProvider : SqlAuthenticationProvider
         /// Instantiates AuthenticationProvider to be used for AAD authentication with MSAL.NET and MSAL.js co-ordinated.
         /// </summary>
         /// <param name="applicationName">Application Name that identifies user folder path location for reading/writing to shared cache.</param>
+        /// <param name="applicationPath">Application Path directory where application cache folder is present.</param>
         /// <param name="authCallback">Callback that handles AAD authentication when user interaction is needed.</param>
-        public AuthenticationProvider(string applicationName)
+        public AuthenticationProvider(string applicationName, string applicationPath)
         {
-            if(string.IsNullOrEmpty(applicationName)) {
+            if (string.IsNullOrEmpty(applicationName))
+            {
                 applicationName = nameof(SqlTools);
+                Logger.Warning($"Application Name not received with command options, using default application name as: {applicationName}");
             }
-            var cachePath = Path.Combine(Utils.BuildAppDirectoryPath(), applicationName, AzureTokenFolder);
+
+            if (string.IsNullOrEmpty(applicationPath))
+            {
+                applicationPath = Utils.BuildAppDirectoryPath();
```

Comment:
Isn't the issue here that the homedir value you're using doesn't seem to be valid for Linux/MacOS?

`var homedir = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);`

https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/8.0/getfolderpath-unix

If you change it to use SpecialFolder.Personal it seems like that should work as well (and match the logic that generates the path being passed from ADS with your current change)

It just seems unnecessary to pass the path like this since the logic on both sides should be the same, and this will still be broken for Linux/MacOS if this parameter isn't passed from ADS it seems like.

Reply:
In future I intend to use this: https://github.com/microsoft/azuredatastudio/blob/9fddefc8fb11fbbd3bb357e11c0e2809d3868aa5/src/vs/platform/environment/node/userDataPath.js#L49

That's why I decided to pass in via ADS.. so we need not worry about managing this path in backend service layer.

Reply:
Yup, good call.

Reply:
Coud you still fix the logic here to use Personal folder? It'll just never work if for some reason the path isn't passed in currently so might as well try to avoid that being a potential issue anywhere.

Reply:
Done, thanks!